docs/book/src/component-config-tutorial/api-changes.md

```diff
@@ -1,5 +1,17 @@
 # Changing things up
 
+<aside class="note warning">
+<h1>Component Config is deprecated</h1>
+
+The ComponentConfig has been deprecated in the Controller-Runtime since its version 0.15.0.  [More info](https://github.com/kubernetes-sigs/controller-runtime/issues/895)
+Moreover, it has undergone breaking changes and is no longer functioning as intended.
+As a result, Kubebuilder, which heavily relies on the Controller Runtime, has also deprecated this feature,
+no longer guaranteeing its functionality from version 3.11.0 onwards. You can find additional details on this issue [here](https://github.com/kubernetes-sigs/controller-runtime/issues/2370).
+
+Please, be aware that it will force Kubebuilder remove this option in soon future release.
```

Comment:
@varshaprasad96 but the idea of the ComponentConfig is we are able to have an option to deploy the same solution (same code) with different config options per env. So, if we pass direct the Options in the manager (default behaviour) then it does not achieve the same goal. Right?

Therefore, should we really say that?

Reply:
Since passing of the configuration options through the config file s removed, the next alternative for it to pass it through `manager.Options`. The idea is letting the user pass all customizable options through manager instead of config file. Which is why it is the alternative to what we have currently. 

We need not block this PR for the docs update, I can add that in a follow up.